lib/gitlab/client/branches.rb

```diff
@@ -61,22 +61,22 @@ def protect_branch(project, branch, options = {})
     # @param  [String] branch The name of the branch.
     # @return [Gitlab::ObjectifiedHash] Details about the branch
     def unprotect_branch(project, branch)
-      delete("/projects/#{url_encode project}/protected_branches/#{url_encode branch}")
+      put("/projects/#{url_encode project}/repository/branches/#{url_encode branch}/unprotect")
```

Comment:
Docs are saying that this API endpoint is deprecated in favor of `DELETE /projects/:id/protected_branches/:name`. https://docs.gitlab.com/ce/api/branches.html#unprotect-repository-branch